Make template variables SSHPublicKey and SSHPrivateKey as strings (#8829)
hashicorp/packer: website/source/docs/templates/engine.html.md

    }
    ```

>     Valid variables to request are: 

Author: @SwampDragons can you check if the descriptions for the variables are correct, please? 🙂

Reviewer: Done!

    - __ID__: Represents the vm being provisioned. For example, in Amazon it is the instance id; in digitalocean, 
    it is the droplet id; in Vmware, it is the vm name.   
     

>     - __Host__, __Port__, __User__ and __Password__: The host, port, user, and password used to access the machine.

Reviewer: ```suggestion
    - __Host__, __Port__, __User__ and __Password__: The host, port, user, and password that Packer uses to access the machine. Useful for using the shell local provisioner to run Ansible or Inspec against the provisioned instance.
```

>     - __Host__, __Port__, __User__ and __Password__: The host, port, user, and password used for access the machine.

Reviewer 2: ```suggestion
    - __Host__, __Port__, __User__ and __Password__: The host, port, user, and password used to access the machine.
```

>     - __ConnType__: Type of the communicator being used. For example, for SSH communicator this will be "ssh".

Reviewer 2: ```suggestion
    - __ConnType__: Type of communicator being used. For example, for SSH communicator this will be "ssh".
```

>     - __PackerRunUUID__: Current build unique id. Can be used to specify build artifacts.

Reviewer 2: ```suggestion
    - __PackerRunUUID__: Current build's unique id. Can be used to specify build artifacts.
```

>     - __PackerHTTPAddr__: Host address `IP:PORT` for the machine.

Reviewer 2: ```suggestion
    - __PackerHTTPAddr__: HTTP address `IP:PORT` to access the host from the VM. To get files for example.
```

>     - __PackerHTTPAddr__: HTTP address `IP:PORT` to access the host from the VM. To get files for example.

Reviewer: ```suggestion
    - __PackerHTTPAddr__: HTTP address of the file server Packer creates to serve items in the "http" dir to the vm, displayed in the format `IP:PORT`.
```

>     - __SSHPublicKey__ and __SSHPrivateKey__: These are unique to the SSH communicator and are unset when using other communicators. 

Reviewer: ```suggestion
    - __SSHPublicKey__ and __SSHPrivateKey__: The public and private key that Packer uses to connect to the instance. These are unique to the SSH communicator and are unset when using other communicators. 
```

>     The values have scape lines and should be wrapped into single quotes to correct format. An example of proper usage:

Reviewer 2: ```suggestion
    - __SSHPublicKey__ and __SSHPrivateKey__: These are unique to the SSH communicator and are unset when using other communicators. 
    __SSHPublicKey__ and __SSHPrivateKey__ can have escape sequences and special characters so their output should be single quoted to avoid surprises.  For example:
```

    Valid variables to request are: "ID", "Host",
    "Port", "User", "Password", "ConnType",

>     "PackerRunUUID", "PackerHTTPAddr", "SSHPublicKey", "SSHPrivateKey", "SSHPublicKeyString", and "SSHPrivateKeyString".

Reviewer 3: Is it documented anywhere that the value of SSHPrivateKey is in bytes? I wonder if we need to document the difference between these two variables.

Author: No, it's not  👀. I'm waiting for @SwampDragons to reply to my question in the issue because maybe I can make the original values as String instead of creating a new one. I don't know when the []byte is useful. Do you know?
If I can do this I can just leave it the way it is, right?

Reviewer 3: Circling back here. I think it is in bytes solely for the input as it comes in from a file and passed around as a byte slice. But as for an output I agree having it as a string is better option. 

Re: I can just leave it the way it is. If you are thinking of changing the input I would say leave that as is and just cast the output to a string.

Author: Actually I wanted just to change the variable type as you said in the end.
